pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1004,12 +1006,26 @@ SegmentBuildDescriptor buildSegmentInternal(boolean forCommit) {
       _segmentLogger.warn("Table data manager is already shut down");
       return null;
     }
+    final long startTimeMillis = now();
     try {
-      final long startTimeMillis = now();
       if (_segBuildSemaphore != null) {
-        _segmentLogger.info("Waiting to acquire semaphore for building segment");
-        _segBuildSemaphore.acquire();
+        _segmentLogger.info("Trying to acquire semaphore for building segment");
+        Instant acquireStart = Instant.now();
+        int timeoutSeconds = 5;
+        while (!_segBuildSemaphore.tryAcquire(timeoutSeconds, TimeUnit.SECONDS)) {
+          _segmentLogger.warn("Could not acquire semaphore for building segment in {}",
+              Duration.between(acquireStart, Instant.now()));
+          timeoutSeconds = Math.min(timeoutSeconds * 2, 300);
+        }
```

Comment:
The initial timeout is 5 seconds but the timeout is being updated to `Math.min(timeoutSeconds * 2, 300)`? Did seconds and minutes get mixed up here or am I missing something?

Reply:
Ah okay, makes sense, I misread / misinterpreted this as `Math.max(timeoutSeconds * 2, 300)` 🤦‍♂️